src/sagas/giphy.js

```diff
@@ -0,0 +1,36 @@
+import { call, takeEvery, put } from 'redux-saga/effects';
+import { getTrendingList, searchGifs, getById } from '../api/giphy';
+import giphyActions from '../actions/giphy';
+
+export function* getTopTrending(records) {
```

Comment:
All of these "fetchers" sagas have a pattern. For a project like this it's enough, but for bigger projects could you come up with an idea to abstract this pattern in order to avoid rewriting the same code for all this type of sagas?

Reply:
Maybe check what is coming in the action and call the adequate method (a la reducers), but, IMO, is better keep those separately because if in one point you need to isolate one of this the structure is ready (yeah, I know is a matter of good planning but I suffered in the past for changes of this kind)